src/ssr.rs

```diff
@@ -1,156 +1,187 @@
-use std::collections::HashMap;
-
-#[derive(Clone, Debug, PartialEq)]
-pub struct Ssr<'a> {
-    // TODO: Check if better Box<str> instead of String
-    source: String,
-    entry_point: &'a str,
+pub struct SSREnvironment {
+    source_code: String,
+    entry_point: String,
+    root_export: String,
+    isolate: v8::OwnedIsolate,
 }
 
-impl<'a> Ssr<'a> {
-    /// Create an instance of the Ssr struct instanciate the v8 platform as well.
-    pub fn new(source: String, entry_point: &'a str) -> Self {
-        Self::init_platform();
-
-        Ssr {
-            source,
-            entry_point,
-        }
-    }
-
-    fn init_platform() {
+impl SSREnvironment {
+    fn init() {
         lazy_static! {
-          static ref INIT_PLATFORM: () = {
-              //Initialize a new V8 platform
-              let platform = v8::new_default_platform(0, false).make_shared();
-              v8::V8::initialize_platform(platform);
-              v8::V8::initialize();
-          };
+            static ref INIT_PLATFORM: () = {
+                //Initialize a new V8 platform
+                let platform = v8::new_default_platform(0, false).make_shared();
+                v8::V8::initialize_platform(platform);
+                v8::V8::initialize();
+            };
         }
 
         lazy_static::initialize(&INIT_PLATFORM);
     }
 
-    /// Evaluates the javascript source code passed as argument and render it as a String.
-    /// Any initial params (if needed) must be passed as JSON.
-    ///
-    /// <a href="https://github.com/Valerioageno/ssr-rs/blob/main/examples/actix_with_initial_props.rs" target="_blank">Here</a> an useful example of how to use initial params with the actix framework.
-    ///
-    /// "enrty_point" is the variable name set from the frontend bundler used. <a href="https://github.com/Valerioageno/ssr-rs/blob/main/client/webpack.ssr.js" target="_blank">Here</a> an example from webpack.
-    pub fn one_shot_render(source: String, entry_point: &str, params: Option<&str>) -> String {
-        Self::init_platform();
+    pub fn new(source_code: &str, entry_point: &str, root_export: &str) -> Self {
```

Comment:
`source_code` should be a `String` since the huge size it could have

Reply:
The argument here being that the "String" would be a copy? Good point. We might want to just keep a reference. The reason I went for a copy here is just so that the user doesn't need to keep the code around for the same lifetime as the SSREnvironment struct but I think it's a valid argument to just keep a reference instead considering some Webpack bundles can be > 5MB.

Reply:
Actually I was wandering that because the runtime would have to coerce the value from `String` to `&str` and again to `String`. That's because `read_to_string() -> String`, then we have to pass it in the function so transform to a `&str` and again create the struct so `String::from()`. Since is a data that we would read just one time at start time it could be better save it on the heap I think.